File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cs
using System.Collections.Generic;

public class Solution
{
    public int[] Intersection(int[] nums1, int[] nums2)
    {
        HashSet<int> set = new HashSet<int>();

        foreach (int num in nums1)
        {
            set.Add(num);
        }

        HashSet<int> result = new HashSet<int>();

        foreach (int num in nums2)
        {
            if (set.Contains(num))
            {
                result.Add(num);
            }
        }

        return new List<int>(result).ToArray();
    }
}

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer>hs=new HashSet<>();
        HashSet<Integer>intersection=new HashSet<>();
        for(int num:nums1){
            hs.add(num);
        }
        for(int nums:nums2){
            if(hs.contains(nums)){
                intersection.add(nums);
            }
        }
        int[] arr=new int[intersection.size()];
        int index=0;
        for(int intersec:intersection){
            arr[index++]=intersec;
        }
        return arr;
    }
}